tests/cbmc/sources/make_common_datastructures.c

```diff
@@ -22,31 +22,35 @@ bool s2n_stuffer_is_bounded(const struct s2n_stuffer *stuffer, const size_t max_
     return (stuffer->blob.size <= max_size);
 }
 
-void ensure_s2n_blob_has_allocated_fields(struct s2n_blob *blob)
+void cbmc_populate_s2n_blob(struct s2n_blob *blob)
 {
-    if (blob->growable) {
-        blob->data = (blob->allocated == 0) ? NULL : malloc(blob->allocated);
-    } else {
-        blob->data = (blob->size == 0) ? NULL : malloc(blob->size);
+    if (blob != NULL) { 
```

Comment:
You do this NULL check for all of them, and it makes every method one loop deeper than it otherwise would be. Maybe add a CBMC version of the ENSURE_REF macro? if(==NULL) { return; }?

Reply:
Good catch!! I applied `PTR_GUARD` whenever possible across all proof allocators. Could you take another look @lrstewart?

Reply:
Would `PTR_GUARD` work here? Wouldn't it return `NULL` from populate functions that should return `void`?

Reply:
Yep, I changed to `CBMC_ENSURE_REF`.